internal/resource-request-operator/offerUpdater.go

```diff
@@ -0,0 +1,175 @@
+	u.queue = workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Offer update queue")
+}
+
+// Start runs the OfferUpdate worker.
+func (u *OfferUpdater) Start(ctx context.Context, group *sync.WaitGroup) {
+	defer u.queue.ShutDown()
+	group.Add(1)
+	defer group.Done()
+	go u.startRunner(ctx)
+	<-ctx.Done()
+}
+
+func (u *OfferUpdater) startRunner(ctx context.Context) {
+	wait.Until(u.run, 2*time.Second, ctx.Done())
+}
+
+func (u *OfferUpdater) run() {
+	for u.processNextItem() {
+	}
+}
+
+func (u *OfferUpdater) processNextItem() bool {
+	obj, shutdown := u.queue.Get()
+	if shutdown {
+		return false
+	}
+	err := func(obj interface{}) error {
+		defer u.queue.Done(obj)
+		var clusterID string
+		var ok bool
+		if clusterID, ok = obj.(string); !ok {
+			// As the item in the workqueue is actually invalid, we call
+			// Forget here else we'd go into a loop of attempting to
+			// process a work item that is invalid.
+			u.queue.Forget(obj)
+			return fmt.Errorf("error getting object %v from OfferUpater queue. It is not a string", obj)
+		}
+		// call createOrUpdate which after some controls generate a new resourceOffer for this clusterID or update it if exists.
+		if requeue, err := u.createOrUpdateOffer(clusterID); err != nil {
+			if requeue {
+				// requeue is true due to a transient error so put the item back on the workqueue.
+				u.queue.AddRateLimited(clusterID)
+			}
+			// requeue == false means that the clusterID is no more valid and so it will be not requeued.
+			u.Remove(clusterID)
+			u.broadcasterInt.RemoveClusterID(clusterID)
+			return fmt.Errorf("error during updating ResourceOffer for cluster %s: %w", clusterID, err)
+		}
+		return nil
+	}(obj)
+	if err != nil {
+		klog.Errorf("Error occurred during ResourceOffer update %s", err)
+		return true
+	}
+	klog.Infof("Update cluster %s processed", obj.(string))
+
+	// requeue after timeout seconds
+	u.queue.AddAfter(obj, getRandomTimeout())
+	return true
+}
+
+func (u *OfferUpdater) createOrUpdateOffer(clusterID string) (bool, error) {
+	list, err := u.getResourceRequest(clusterID)
+	if err != nil {
+		return true, err
+	} else if len(list.Items) != 1 {
+		// invalid clusterID so return requeue = false. The clusterID will be removed from the workqueue and broadacaster maps.
+		return false, fmt.Errorf("ClusterID %s is no more valid. Deleting", clusterID)
+	}
+	request := list.Items[0]
+	resources := u.broadcasterInt.ReadResources(clusterID)
+	offer := &sharingv1alpha1.ResourceOffer{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: request.GetNamespace(),
+			Name:      offerPrefix + u.homeClusterID,
+		},
+	}
+
+	op, err := controllerutil.CreateOrUpdate(context.Background(), u.Client, offer, func() error {
+		offer.Labels = map[string]string{
+			discovery.ClusterIDLabel:         request.Spec.ClusterIdentity.ClusterID,
+			crdreplicator.LocalLabelSelector: "true",
+			crdreplicator.DestinationLabel:   request.Spec.ClusterIdentity.ClusterID,
+		}
```

Comment:
Should not overwrite the entire map, but just add your owns to keep user-added ones.